front/src/resizable_drawer.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { Drawer, DrawerProps } from '@mui/material';
+import { useEffect } from 'react';
+
+type ResizableDrawerProps = {
+    minWidth: number;
+    maxWidth: number;
+    children: React.ReactNode;
+    drawerProps?: DrawerProps;
+    onWidthChange?: (width: number) => void;
+};
+
+const INITAL_WIDTH = 300;
+
+export function ResizableDrawer(props: ResizableDrawerProps) {
+    const [width, setWidth] = React.useState(INITAL_WIDTH);
+
+    useEffect(() => {
+        if (props.onWidthChange) {
```

Comment:
I think this one should also work - `props.onWidthChange && props.onWidthChange(width);`